backend/src/bundles/chat/chat.service.ts

```diff
@@ -0,0 +1,64 @@
+import { type ValueOf } from 'shared';
+import { type Tiktoken, encoding_for_model } from 'tiktoken';
+
+import { CHAT_MODEL } from '~/common/services/open-ai/libs/constants/constants.js';
+import { type OpenAIRole } from '~/common/services/open-ai/libs/enums/enums.js';
+
+import {
+    type ChatService as ChatServiceT,
+    type Message,
+} from './libs/types/types.js';
+
+class ChatService implements ChatServiceT {
+    private modelEncoding: Tiktoken;
+
+    public constructor() {
+        this.modelEncoding = encoding_for_model(CHAT_MODEL);
+    }
+
+    public addMessageToHistory(
+        chatHistory: Message[],
+        userMessage: string,
+        role: ValueOf<typeof OpenAIRole>,
+    ): void {
+        const newUserMessage = {
+            content: userMessage,
+            role,
+        };
+
+        chatHistory.push(newUserMessage);
```

Comment:
You return new messages array in deleteOldMessages method, but you still mutate chatHistory parameter here

Reply:
Done